decompiler_agent/src/main/java/org/jrd/agent/api/Variables.java

```diff
@@ -0,0 +1,95 @@
+package org.jrd.agent.api;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Variables {
+
+    /**
+     * we have to have all inernal classes inited, so any later classes can use them without manual INIT.
+     */
+    public static void init() {
+        NoSuchFakeFieldException.init();
+        Global.init();
+    }
+
+    protected Variables() {
+        // prevents calls from subclass... checkstile sucks
+        throw new UnsupportedOperationException();
+    }
+
+    public static class NoSuchFakeFieldException extends RuntimeException {
+
+        public NoSuchFakeFieldException() {
+        }
+
+        public NoSuchFakeFieldException(String s) {
+            super(s);
+        }
+
+        public static void init() {
+        }
+    }
+
+    public static class Global {
+
+        protected Global() {
+            // prevents calls from subclass... checkstile sucks
+            throw new UnsupportedOperationException();
+        }
+
+        private static final Map<String, Object> GLOBALS = Collections.synchronizedMap(new HashMap<>());
+
+        public static void init() {
+        }
+
+        public static Object set(String name, Object value) {
+            Object old = GLOBALS.put(name, value);
+            if ("jrd.debug".equals(old)) {
+                System.err.println("We do nto care about old");
```

Comment:
You know we have AgentLogger for this. Besides, what is the meaning of this if-statement? Why is it there?

Reply:
I wanted to have cleat, we are not returning old which may be expected. But check style did nto allwoed unused field...

Reply:
> we are not returning old which may be expected

But... there is no separate return statement in the if. We are returning old, either way.

Reply:
what?

Reply:
no. value is returned eitherway.